design/client_conformance_profiles.md

```diff
@@ -0,0 +1,550 @@
+If `private_key_jwt`, the signature algorithm of JWS Client Assertion is `ES256`, `ES384`, `ES512`, `PS256`, `PS384` or `PS512`.
+
+* Reference:
+    - Authz Server / 14, 14-1, 14-2
+    - JWS algorithm considerations / 1, 2, 3
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+##### 2. Holder-of-Key Token
+In TLS handshake, the client is authenticated by its PKI X.509 Certificate.
+
+Keycloak binds this certificate with an access token and a refresh token by following Mutual-TLS Client Certificate-Bound Access Tokens [MTLS].
+
+* Reference:
+    - Authz Server / 5, 6
+    - Public Client / 1
+    - Confidential Client / 1
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+#### Endpoint : Token Introspection Endpoint
+
+Client's Request Check
+
+* Operations :
+    - handle token introspection request
+
+##### 1. Client Authentication Method
+The client challenges its authentication by the method pre-registered in `token_endpoint_auth_method` of OpenID Connect Client Metadata (`tls_client_auth_subject_dn` or `private_key_jwt`).
+
+If `private_key_jwt`, the signature algorithm of JWS Client Assertion is `ES256`, `ES384`, `ES512`, `PS256`, `PS384` or `PS512`.
+
+* Reference:
+    - Authz Server / 14, 14-1, 14-2
+    - JWS algorithm considerations / 1, 2, 3
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+#### Endpoint : Logout Endpoint
+
+Client's Request Check
+
+* Operations :
+    - handle (backchannel) logout request
+
+##### 1. Client Authentication Method
+The client challenges its authentication by the method pre-registered in `token_endpoint_auth_method` of OpenID Connect Client Metadata (`tls_client_auth_subject_dn` or `private_key_jwt`).
+
+If `private_key_jwt`, the signature algorithm of JWS Client Assertion is `ES256`, `ES384`, `ES512`, `PS256`, `PS384` or `PS512`.
+
+* Reference:
+    - Authz Server / 14, 14-1, 14-2
+    - JWS algorithm considerations / 1, 2, 3
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+##### 2. Holder-of-Key Token
+In TLS handshake, the client is authenticated by its PKI X.509 Certificate.
+
+Keycloak verifies whether the refresh token is bound with this certificate by following Mutual-TLS Client Certificate-Bound Access Tokens [MTLS].
+
+* Reference:
+    - Authz Server / 5, 6
+    - Public Client / 1
+    - Confidential Client / 1
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+#### Endpoint : UserInfo Endpoint
+
+Client's Request Check
+
+* Operations :
+    - handle UserInfo request
+
+##### 1. Holder-of-Key Token
+In TLS handshake, the client is authenticated by its PKI X.509 Certificate.
+
+Keycloak verifies whether the access token is bound with this certificate by following Mutual-TLS Client Certificate-Bound Access Tokens [MTLS].
+
+* Reference:
+    - Authz Server / 5, 6
+    - Public Client / 1
+    - Confidential Client / 1
+
+Error Response : 400 Bad Request (the detail is TBD)
+
+## TODO
+
+### "acr" (Authentication Context Class Reference) treatment

```

Comment:
We hope to start on step-up authentication later this year. But as usual, no promise for anything :) Help from the community with the feedback and implementation will be more than welcome!

Reply:
@mposolda I've recognized that this "acr" matter has been opened an discussed in [KEYCLOAK-3314 of JIRA ticket](https://issues.redhat.com/browse/KEYCLOAK-3314) and not yet resolved.

I've looked into release-v3.0.3 of the [FAPI conformance test](https://gitlab.com/openid/conformance-suite) and found that it seems not test "acr" values. Therefore, I've left "acr" matter TODO at this time.

However, I hope I would like to resolve this "acr" matter along with your step-up authentication.

If this "acr" matter need to be discussed, [its JIRA ticket's comment](https://issues.redhat.com/browse/KEYCLOAK-3314) is appropriate?

Reply:
I see, I agree with you. So what about the following?

* when the admin changes the client conformance profile but some settings violates against the new profile, show the separate UI to show the recommended settings per profile.